src/App.js

```diff
@@ -1,19 +1,295 @@
+    .then((response) => {
+      setMovies(response.data.data);
+    })
+    .catch((err) => {
+      console.log('error', err)
+     });
+  }
+
+  const createSearchParamsCustom = ()=>{
+    const queryParams ={activeGenre: selectedGenre, sortBy: sortBy};
+    if(searchString){
+      queryParams['query'] = searchString;
+    }
+    return createSearchParams(queryParams);
+  }
+  const onGenreSelected = (selectedGenre) => {
+    console.log("You selected: "+ selectedGenre)
+    setSearchString(null)
+    setSelectedGenre(selectedGenre);
+    navigate({
+      pathname: "/",
+      search: createSearchParamsCustom().toString()
+    });
+  }
+
+  const onTileSelected = (movieData) => {
+    navigate("/movie/"+movieData.id)
+    setSelectedMovieData(movieData)
+    toggleMovieInfoPanel(true)
+  }
+
+  const toggleMovieInfoPanel = (show) =>{
+    window.scrollTo(0, 0)
+    setShowMovieInfoPanel(show)
+    console.log("toggleMovieInfoPanel called: ", show)
+  }
+
+  const viewMovieData = (movieData) => {
+    console.log("viewMovieData: ", movieData)
+    // alert("viewMovieData: "+ movieData)
+    var str="\n\n";
+    for (const [key, value] of Object.entries(movieData)) {
+      str = str + key+ " : "+ value + "\n";
+    }
+    alert("=====viewMovieData======="+ str)
+  }
+
+  const addMovieData = () => {
+    navigate("/movie/add")
+  }
+  
+  const editMovieData = (movieData) => {
+    console.log("editMovieData: ", movieData)
+    // alert("editMovieData: "+ movieData)
+    var str="\n\n";
+    for (const [key, value] of Object.entries(movieData)) {
+      str = str + key+ " : "+ value + "\n";
+    }
+    // alert("=====editMovieData======="+ str)
+    
+    navigate("/"+movieData.id+"/edit?"+fetchCurrentQueryParams(searchParams))
+  }
+
+  const deleteMovieData = (movieData) => {
+    console.log("deleteMovieData: ", movieData)
+    // alert("deleteMovieData: "+ movieData)
+    var str="\n\n";
+    for (const [key, value] of Object.entries(movieData)) {
+      str = str + key+ " : "+ value + "\n";
+    }
+    // alert("=====deleteMovieData======="+ str)
+    openDeleteMovieDialog(movieData)
+  }
+
+  const onSortSelectionListener = (sortBy) =>{
+    console.log("onSortSelectionListener", sortBy)
+    setSortBy(sortBy);
+    navigate({
+      pathname: "/",
+      search: createSearchParamsCustom().toString()
+    });
+  }
+
+  const onSearch = (inputText) =>{
+    setSearchString(inputText);
+    callMoviewAPI(null, sortBy, inputText);
+  }
+
+  
+
+  const onDeleteMovie = (movieInfo) =>{
+    console.log(movieInfo)
+    axios.delete(HOST+"/movies/"+movieInfo.id)
+    .then((response) => {
+      openMovieFormResultDialog("The movie has been deleted from database successfully", true)
+    })
+    .catch((err) => {
+      console.log('error', err)
+      openMovieFormResultDialog("Failed to delete the movie. Please try again !!", false)
+     }).finally(()=>{
+      closeDeleteMovieDialog()
+     });
+  }
+
```

Comment:
This too should be in a service file, and again use async await.

For any static strings such as "The movie has been deleted from database successfully", "/movies/", 'error', "Failed to delete the movie. Please try again !!" use constants. And for strings which are shown to the user, you might want to read about internationaliztion concept (not of high importance though).